mysql/azuremysql/main.tf

```diff
@@ -0,0 +1,94 @@
+
+terraform {
+  required_version = "~>0.11"
+}
+
+provider "azurerm" {  
+  version = "~> 1.22"
+  # See documentation for more info: https://www.terraform.io/docs/providers/azurerm/auth/azure_cli.html
+  subscription_id = "00000000-0000-0000-0000-000000000000"
+}
+provider "random" {
+  version = "~> 2.1"
+}
+
+# Run Azure CLI command "az account list-locations" to see list of all locations.
+variable "location" {
+  description = "The Azure Region in which all resources in this example should be created."
+}
+# See documentation for more info: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview
+variable "resourcegroup" {
+  description = "The Azure Resource Group Name within your Subscription in which this resource will be created."
+}
+variable "servername" {
+  description = "The MySql Server Name, must be unique."
+}
+
+resource "random_string" "db_password" {
+  keepers = {
+    region = "${var.location}"
+  }
+
+  special = false
+  length  = 20
+}
+resource "azurerm_resource_group" "mysqlrg" {
+  name     = "${var.resourcegroup}"
+  location = "${var.location}"
+}
+resource "azurerm_mysql_server" "mysqlserver" {
+  name                = "${var.servername}"
```

Comment:
If this must be unique, then generating a random suffix for something like "go-cdk-test-" seems easier for the user rather than prompting for it.

Reply:
Ok, servername is auto generated using your suggested prefix.